src/lib/api.js

```diff
@@ -631,7 +645,40 @@ const searchItems = async function (collectionId, queryParameters, backend, endp
         results: []
       }
     } else {
-      throw error
+      try {
+        // Parse some types of geometry errors from Search
+        // @ts-ignore
+        const reason1 = error['meta']['body']['error']['caused_by'][0]['reason']
+        return {
+          statusCode: 400,
+          body: reason1
+        }
+      } catch (e1) {
+        // Parse other types of geometry errors from Search
+        try {
+          if (JSON.stringify(error).includes('failed to create query')) {
+            // @ts-ignore
+            const reason2 = error['meta']['body']['error']['root_cause'][0]['reason']
+            return {
+              statusCode: 400,
+              body: reason2
+            }
+          }
+        } catch (e2) {
+          // Flatten response, looking for generic Search error verbage
+          try {
+            if (JSON.stringify(error).includes('failed to create query')) {
+              return {
+                statusCode: 400,
+                body: 'Query failed. Please verify you are submitting a queryable payload.'
+              }
+            }
```

Comment:
`error` is sometimes stringified twice. This isn't a critical path so not a required fix.

```suggestion
        const failedToCreateQuery = JSON.stringify(error).includes('failed to create query');
        // Parse other types of geometry errors from Search
        try {
          if (failedToCreateQuery) {
            // @ts-ignore
            const reason2 = error['meta']['body']['error']['root_cause'][0]['reason']
            return {
              statusCode: 400,
              body: reason2
            }
          }
        } catch (e2) {
          // Flatten response, looking for generic Search error verbage
          try {
            if (failedToCreateQuery) {
              return {
                statusCode: 400,
                body: 'Query failed. Please verify you are submitting a queryable payload.'
              }
            }
```